kubernetes/linux/setup.sh

```diff
@@ -11,6 +13,20 @@ fi
 #upgrade apt to latest version
 apt-get update && apt-get install -y apt && DEBIAN_FRONTEND=noninteractive apt-get install -y locales
 
+
+curl -sSL https://rvm.io/mpapis.asc | gpg --import -
+curl -sSL https://rvm.io/pkuczynski.asc | gpg --import -
+curl -sSL https://get.rvm.io | bash -s stable
+
+# setup paths for ruby and rvm
+if [ -f /etc/profile.d/rvm.sh ]; then
+    source /etc/profile.d/rvm.sh
+    echo "[ -f /etc/profile.d/rvm.sh ] && source /etc/profile.d/rvm.sh" >> ~/.bashrc
+fi
+
+rvm install 3.1.0
```

Comment:
@pfrcks , In ruby 3.1, Yajl JSON gem is root cause for the data gaps and CPU throttling on scale cluster.  This issue easily reproducible with 2K pods on the cluster or running test ruby app (within container) parsing ~2MB JSON payload. Default JSON parser addresses this issue but we would need more testing as this JSON parser in critical path and doesn't cause any perf regressions. Lets spend more time on testing once your are back.  If all the tests pass, then we get this change in next agent release.